src/tds/data.c

```diff
@@ -718,6 +717,15 @@ tds_generic_get(TDSSOCKET * tds, TDSCOLUMN * curcol)
 	tdsdump_log(TDS_DBG_INFO1, "tds_get_data: type %d, varint size %d\n", curcol->column_type, curcol->column_varint_size);
 	switch (curcol->column_varint_size) {
 	case 4:
+		if (!is_blob_type(curcol->column_type)) {
```

Comment:
If types are blob `column_varint_size` is 5, not 4, so this code is never executed.